### PR TITLE
adapt to new partial PosixDate.format

### PR DESCRIPTION
**Pull request by mfelsche**
introduced with https://github.com/ponylang/ponyc/pull/2971

### CODE REVIEW

**Comment by mfelsche** [comment]
So, I did as you suggested @SeanTAllen but with a tiny twist:
My idea is to test PRs and the master branch (nightly) against ponyc master
and have the release branch, from which we do the releases (similarly to how it is set up in ponyc) tested nightly against ponyc release.

This way i'd like to ensure that the master branch is always compatible with the ponyc master branch and that the latest release is compatible with the latest ponyc release.

This somehow answers #7 . I would like to get rid of an extra release branch and all this cruft around it, but for the sake of ensuring compatibility with ponyc (and because it already works that way and i just had to adapt the CI configs) i think going with a release branch is indeed the right thing here.

File: http/common_log.pony [hunk 1]
```diff
@@ -22,10 +22,12 @@ class CommonLog is Logger
     list.push(" - ")
     list.push(_entry(request.url.user))
 
-    let time = PosixDate(Time.seconds()).format("%d/%b/%Y:%H:%M:%S +0000")
-    list.push(" [")
-    list.push(time)
-    list.push("] \"")
+    try // this one should never fail
```
Comment (jemc) [inline http/common_log.pony]: I won't hold up the PR on this account, but I don't think we can truly be confident in this not failing unless we have a test in the test suite that exercises this and proves that the format string works at least once. I didn't see such a test in the test suite - did I miss it?

Comment (SeanTAllen) [reply http/common_log.pony]: i couldnt find either

Comment (mfelsche) [reply http/common_log.pony]: i added a test for this.

**Pull request merged**